operator/internal/controller/redpanda/redpanda_node_pvc_controller.go

```diff
@@ -124,46 +117,38 @@ func (r *RedpandaNodePVCReconciler) reconcile(ctx context.Context, req ctrl.Requ
 			errs = errors.Join(errs, fmt.Errorf("worker node annotation not found or node name is empty: %q", workerNode)) //nolint:goerr113 // joining since we do not error here
 		}
 
-		// we are not being deleted, move on
+		// K8S Node label from PVC does not matches delete K8S Node event
 		if workerNode != req.Name {
 			continue
 		}
 
-		// Now check if we are allowed to operate on this pvc as there may be another controller working
-		var releaseName string
-		if val, ok := pvc.Labels[K8sInstanceLabelKey]; !ok || !isValidReleaseName(val, redpandaNameList) {
-			Infof(log, "could not find instance label or unable retrieve valid releaseName: %s", val)
-			continue
-		} else {
-			releaseName = val
-		}
+		val := pvc.Labels[K8sInstanceLabelKey]
+
+		if r.OperatorMode {
+			rp, ok := redpandasMap[val]
+			if !ok {
+				Infof(log, "could not find Redpanda %q", val)
+				continue
+			}
 
-		// we are in operator mode, check if another controller has ownership here
-		// we will silently fail where we cannot find get values, parse the values file etc..
-		if r.OperatorMode { // nolint:nestif // this is ok
-			doTask, foundKey := doTaskMap[releaseName]
-			if foundKey && !doTask {
+			// Now check if we are allowed to operate on this pvc as there may be another controller working
+			if rp.Spec.ClusterSpec.Statefulset != nil &&
+				rp.Spec.ClusterSpec.Statefulset.SideCars != nil &&
+				rp.Spec.ClusterSpec.Statefulset.SideCars.Controllers != nil &&
+				ptr.Deref(rp.Spec.ClusterSpec.Statefulset.SideCars.Controllers.Enabled, false) {
+				Infof(log, "operator can not reconcile as side-car must reconcole node delete event (%q)", req.Name)
 				continue
 			}
+		} else {
+			releaseName, ok := os.LookupEnv(EnvHelmReleaseNameKey)
```

Comment:
To make this logic a bit more understandable, could we hoist this block our of this loop and exit early?

```go
releaseName, ok := os.LookupEnv(...)

if !ok && !r.OperatorMode {
// Invalid side car mode configuration. Log and bail out early?
}
```